astropy/io/misc/asdf/tags/coords/coords.py

```diff
@@ -0,0 +1,55 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from asdf.yamlutil import custom_tree_to_tagged_tree
+
+from astropy.units import Quantity
+from astropy.coordinates import ICRS, Longitude, Latitude, Angle
+
+from ...types import AstropyType
+from ..unit.quantity import QuantityType
+
+
+__all__ = ['ICRSCoordType']
+
+
+class ICRSCoordType(AstropyType):
+    name = "coords/icrs_coord"
+    types = ['astropy.coordinates.ICRS']
+    requires = ['astropy']
+    version = "1.0.0"
+
+    @classmethod
+    def from_tree(cls, node, ctx):
+        angle = QuantityType.from_tree(node['ra']['wrap_angle'], ctx)
+        wrap_angle = Angle(angle.value, unit=angle.unit)
+        ra = Longitude(
+            node['ra']['value'],
+            unit=node['ra']['unit'],
+            wrap_angle=wrap_angle)
+        dec = Latitude(node['dec']['value'], unit=node['dec']['unit'])
+
+        return ICRS(ra=ra, dec=dec)
+
+    @classmethod
+    def to_tree(cls, frame, ctx):
+        node = {}
+
+        wrap_angle = Quantity(
```

Comment:
Similarly, this can be `Quantity(frame.ra.wrap_angle)` or `frame.ra.wrap_angle.view(Quantity)`